lua/telescope/actions/init.lua

```diff
@@ -805,6 +805,213 @@ actions.cycle_previewers_prev = function(prompt_bufnr)
+
+  local mappings = {}
+  for _, v in pairs(prompt_buf_mappings) do
+    if type(v) == "table" then
+      local name = ""
+      for _, action in ipairs(v) do
+        if type(action) == "string" then
+          name = name == "" and action or name .. " + " .. action
+        end
+      end
+      table.insert(mappings, { mode = v.mode, keybind = v.keybind, name = name })
+    end
+  end
+  if opts.only_show_current_mode then
+    local mode = vim.api.nvim_get_mode().mode
+    mappings = vim.tbl_filter(function(t)
+      return t.mode == mode
+    end, mappings)
+  end
+
+  table.sort(mappings, function(x, y)
+    if x.name < y.name then
+      return true
+    elseif x.name == y.name then
+      -- show normal mode as the standard mode first
+      if x.mode > y.mode then
+        return true
+      else
+        -- loosely prefers simpler mappings
+        if x.keybind < y.keybind then
+          return true
+        end
+        return false
+      end
+    else
+      return false
+    end
+  end)
+
+  local entry_width = #opts.column_padding
+    + opts.mode_width
+    + opts.keybind_width
+    + opts.name_width
+    + (3 * #opts.separator)
+  local num_total_columns = math.floor((vim.o.columns - #opts.column_indent) / entry_width)
+  opts.num_rows = math.ceil(math.min(#mappings / num_total_columns, opts.max_height * vim.o.lines))
+  local total_available_entries = opts.num_rows * num_total_columns
+
+  km_buf = vim.api.nvim_create_buf(false, false)
+  vim.api.nvim_buf_set_name(km_buf, "_TelescopePickersKeymap")
+
+  --   place top or bottom
+  local prompt_win = action_state.get_current_picker(prompt_bufnr).prompt_win
+  local prompt_row = vim.api.nvim_win_get_position(prompt_win)[1]
+  local prompt_pos = prompt_row <= 0.4 * vim.o.lines
+  local km_winid = vim.api.nvim_open_win(km_buf, false, {
+    relative = "editor",
+    anchor = "NW",
+    width = vim.o.columns,
+    height = opts.num_rows + 2 * opts.line_padding,
+    row = prompt_pos == true and vim.o.lines - (opts.num_rows + opts.line_padding) or 0,
+    col = 1,
+    style = "minimal",
+    noautocmd = true,
+  })
+  vim.cmd(
+    "autocmd BufLeave <buffer> ++once lua pcall(vim.api.nvim_win_close, "
+      .. km_winid
+      .. ", true); pcall(vim.api.nvim_buf_delete, "
+      .. km_buf
+      .. ", {force=true})"
+  )
+
+  vim.api.nvim_buf_set_lines(km_buf, 0, -1, false, utils.repeated_table(opts.num_rows + 2, opts.column_indent))
+
+  local keymap_highlights = vim.api.nvim_create_namespace "telescope_keymaps"
+  local highlights = {}
+  for index, mapping in ipairs(mappings) do
+    local row = utils.cycle(index, opts.num_rows) - 1 + opts.line_padding
+    local prev_line = vim.api.nvim_buf_get_lines(km_buf, row, row + 1, false)[1]
+    if index == total_available_entries then
+      local new_line = prev_line .. "..."
+      vim.api.nvim_buf_set_lines(km_buf, row, row + 1, false, { new_line })
+      break
+    end
```

Comment:
With this implementation, if you have exactly the same number of mappings as available spaces, you don't see all of them. Maybe add a check here if it is the last mapping?

Reply:
:sweat_smile: